src/node/db/API.js

```diff
@@ -298,15 +298,93 @@ exports.setText = function(padID, text, callback)
   getPadSafe(padID, true, function(err, pad)
   {
     if(ERR(err, callback)) return;
-    
+
     //set the text
     pad.setText(text);
-    
+
     //update the clients on the pad
     padMessageHandler.updatePadClients(pad, callback);
   });
 }
 
+exports.insertText = function(padID, newText, lineNum, authorMapper, callback)
+{
+  //text is required
+  if(typeof newText != "string")
+  {
+    callback(new customError("text is no string","apierror"));
+    return;
+  }
+
+  //check if lineNum is a number
+  if(!isNaN(parseInt(lineNum)))
+  {
+    lineNum = parseInt(lineNum);
+  }
+  else
+  {
+    callback(new customError("lineNum not given","apierror"));
+    return;
+  }
+
+  if(typeof authorMapper != "string")
+  {
+    callback(new customError("text is no authorMapper","apierror"));
+    return;
+  }
+
+  var authorMangager = require("./AuthorManager");
+  // Check author exists
+  authorMangager.createAuthorIfNotExistsFor(authorMapper, false, function(err, authorID)
+  {
+    if(ERR(err, callback)) return;
+
+    //author does not exist
+    if(authorID == false)
+    {
+      callback(new customError("Author not given or given author does not exist","apierror"));
+      return;
+    }
+    else
+    {
+      //get the pad
+      getPadSafe(padID, true, function(err, pad)
+      {
+        if(ERR(err, callback)) return;
+        //set the text
+        var Changeset = require("ep_etherpad-lite/static/js/Changeset");
+        var oldText = pad.text();
+        // Index of insert point
+        var oldLines = oldText.split("\n");
+        var insertPoint = 0;
+        if(lineNum < oldLines.length) {
+          for (var i = 0; i < lineNum-1; i++) {
+            insertPoint += oldLines[i].length + 1;
+          }
+        }
+        else
+        {
+          insertPoint = oldLen;
```

Comment:
This seems to be the reason why you get an error if you specify a lineNum that is higher than the actual number of lines on this pad